reconcile/utils/rosa/rosa_cli.py

```diff
@@ -0,0 +1,193 @@
+
+
+class RosaCliException(Exception, RosaCliResult):
+    """
+    Represents an exception that occurred during a ROSA CLI execution.
+    """
+
+    def __init__(
+        self,
+        status: JobStatus,
+        command: str,
+        log_handle: Optional[LogHandle] = None,
+    ) -> None:
+        Exception.__init__(
+            self, f"ROSA CLI execution failed with status: {status}, cmd: {command}"
+        )
+        RosaCliResult.__init__(self, status, command, log_handle)
+
+
+class RosaJob(K8sJob, BaseModel, frozen=True):
+    """
+    Represents a ROSA CLI job. It leverages the reconcile.utils.jobcontroller module
+    functionality to execute ROSA CLI commands in a Kubernetes cluster.
+
+    Since the ROSA CLI requires access to both AWS and OCM, the job is executed
+    with the required credentials and tokens.
+
+    The credentials used for AWS are as temporary ones that expire after a
+    defined amount of time, usually 15 minutes. The OCM access token is provided
+    as a JWT with an expiration of 15 minutes.
+
+    Since both credential types are of temporary nature, they are passed to the
+    Kubernetes job as environment variables.
+    """
+
+    account_name: str
+    cluster_name: str
+    org_id: str
+    cmd: str
+
+    aws_credentials: AWSTemporaryCredentials
+    ocm_token: str
+
+    dry_run: bool = False
+
+    def name_prefix(self) -> str:
+        prefix = "rosa-cli"
+        if self.dry_run:
+            prefix += "-dry-run"
+        return prefix
+
+    def unit_of_work_identity(self) -> Any:
+        return {
+            "cmd": self.cmd,
+            "account_name": self.account_name,
+            "cluster_name": self.cluster_name,
+            "org_id": self.org_id,
+            "dry_run": self.dry_run,
+        }
+
+    def annotations(self) -> dict[str, str]:
+        return {
+            "qontract.rosa.account_name": self.account_name,
+            "qontract.rosa.cluster_name": self.cluster_name,
+            "qontract.rosa.org_id": self.org_id,
+        }
+
+    def job_spec(self) -> V1JobSpec:
+        return V1JobSpec(
+            backoff_limit=1,
+            ttl_seconds_after_finished=3600,
+            template=V1PodTemplateSpec(
+                metadata=V1ObjectMeta(
+                    annotations=self.annotations(), labels=self.labels()
+                ),
+                spec=V1PodSpec(
+                    containers=[
+                        V1Container(
+                            name="outputs",
+                            image="registry.ci.openshift.org/ci/rosa-aws-cli:latest",
+                            command=["/bin/bash", "-c"],
+                            args=[self.cmd],
+                            image_pull_policy="Always",
+                            env=[
+                                V1EnvVar(
+                                    name="AWS_ACCESS_KEY_ID",
+                                    value=self.aws_credentials.access_key_id,
```

Comment:
can we have the creds as a secret and change these to a secretRef? As it is a user with view privs on the namespace would be able to get those secrets from the job/pod definitions

Reply:
for automated cleanup we can set an `ownerReference` on the secret such that it gets garbage collected when the job is deleted after `ttl_seconds_after_finished`

Reply:
the general idea was to use temporary credentials exclusively, so these AWS creds and also the OCM creds are invalid after 15 minutes (15 min is the minimum valididity for AWS STS tmp creds and 15 min is also the validity of an OCM access token).

i considered secrets but it implied managing the secret resource as well, which makes the job controller more complex again. the benefit of having only a job resource revolves around the `ttl_seconds_after_finished` property and its ability to clean up "old" jobs. 

i'm not saying managing a secret is unreasonable efforts. but i tend to believe that temporary credentials that expire after 15 minutes combined with a namespace where only we have access should be sufficient protection.

i'm not going to die on that hill though. if you have serious concerns security wise, we can discuss a design that revolves around secrets.

Reply:
That was what I wanted to do too, using secretRef, however considering the STS temporary creds are only valid for short period of time which largely limited the risk. I think it's okay to use env.

Reply:
I still think the idea of using `ownerReferences` is interesting and would solve this security concern but I understand this adds complexity where it may not be needed due to the ephemeral nature of the credentials. I'm not going to die on that hill either :)

Reply:
hmm. would `ownerReference` work?  what would be the order of resource creation?

a) create the job that references the secret that does not exist yet, then create the secret with the job in the `ownerReference`. would the job pod wait for the secret to appear?

b) create the secret, without an owner reference, create the job, patch the owner ref in the secret (does that work)?

Reply:
i gave it a try and a) seems to work. i will implement it like this. thanks for your feedback on this aspect